src/main/kotlin/com/terraformation/backend/accelerator/db/ParticipantProjectSpeciesStore.kt

```diff
@@ -78,6 +79,31 @@ class ParticipantProjectSpeciesStore(
     }
   }
 
+  fun deleteMany(participantProjectSpeciesIds: Set<ParticipantProjectSpeciesId>) {
+    participantProjectSpeciesIds.forEach {
+      requirePermissions { deleteParticipantProjectSpecies(it) }
+    }
+
+    dslContext.transactionResult { _ ->
+      val rowsDeleted =
+          dslContext
+              .deleteFrom(PARTICIPANT_PROJECT_SPECIES)
+              .where(PARTICIPANT_PROJECT_SPECIES.ID.`in`(participantProjectSpeciesIds))
+              .execute()
+
+      if (rowsDeleted != participantProjectSpeciesIds.size) {
+        val failed =
+            dslContext
+                .select(PARTICIPANT_PROJECT_SPECIES.ID)
+                .from(PARTICIPANT_PROJECT_SPECIES)
+                .where(PARTICIPANT_PROJECT_SPECIES.ID.`in`(participantProjectSpeciesIds))
+                .fetchSet(PARTICIPANT_PROJECT_SPECIES.ID.asNonNullable())
```

Comment:
Good catch

Reply:
The idea here is to verify that the IDs were deleted. Elsewhere I see to just check the row count and throw an error if it isn't what is expected. In this case I thought it would be nice to indicate which IDs failed to delete, but it seems like that would require a select before the delete to see which IDs wouldn't be deleted. Is that worth it, or am I adding unnecessary complexity?

Reply:
Probably unnecessary complexity, I agree. You _can_ do it in one query (the `RETURNING` clause lets you specify a list of columns to return from affected rows on a write operation) but probably not worth the bother.

Reply:
I got it to work as expected, and satisfy the test case below, but will remove for simplicity.